src/test/app/XChain_test.cpp

```diff
@@ -0,0 +1,5120 @@
+        // fail
+        XEnv(*this)
+            .tx(create_bridge(mcDoor, jvb))
+            .fund(res0 + one_xrp - drops(1), mcuAlice)
+            .close()
+            .tx(xchain_commit(mcuAlice, jvb, 1, one_xrp, scBob),
+                ter(tecUNFUNDED_PAYMENT));
+
+        auto const minAccountCreate = XRP(20);
+
+        // account create commit where the fee dips into the reserve,
+        // this should succeed
+        XEnv(*this)
+            .tx(create_bridge(mcDoor, jvb, reward, minAccountCreate))
+            .fund(
+                res0 + tx_fee + minAccountCreate + reward - drops(1), mcuAlice)
+            .close()
+            .tx(sidechain_xchain_account_create(
+                    mcuAlice, jvb, scuAlice, minAccountCreate, reward),
+                ter(tesSUCCESS));
+
+        // account create commit where the commit dips into the reserve,
+        // this should fail
+        XEnv(*this)
+            .tx(create_bridge(mcDoor, jvb, reward, minAccountCreate))
+            .fund(res0 + minAccountCreate + reward - drops(1), mcuAlice)
+            .close()
+            .tx(sidechain_xchain_account_create(
+                    mcuAlice, jvb, scuAlice, minAccountCreate, reward),
+                ter(tecUNFUNDED_PAYMENT));
+    }
+
+    void
+    testXChainDeleteDoor()
+    {
+        using namespace jtx;
+
+        testcase("Bridge Delete Door Account");
+
+        auto const acctDelFee{
+            drops(XEnv(*this).env_.current()->fees().increment)};
+
+        // Deleting an account that owns bridge should fail
+        {
+            XEnv mcEnv(*this);
+
+            mcEnv.tx(create_bridge(mcDoor, jvb, XRP(1), XRP(1))).close();
+
+            // We don't allow an account to be deleted if its sequence
+            // number is within 256 of the current ledger.
+            for (size_t i = 0; i < 256; ++i)
+                mcEnv.close();
+
+            // try to delete mcDoor, send funds to mcAlice
+            mcEnv.tx(
+                acctdelete(mcDoor, mcAlice),
+                fee(acctDelFee),
+                ter(tecHAS_OBLIGATIONS));
+        }
+
+        // Deleting an account that owns a claim id should fail
+        {
+            XEnv scEnv(*this, true);
+
+            scEnv.tx(create_bridge(Account::master, jvb))
+                .close()
+                .tx(xchain_create_claim_id(scAlice, jvb, reward, mcAlice))
+                .close();
+
+            // We don't allow an account to be deleted if its sequence
+            // number is within 256 of the current ledger.
+            for (size_t i = 0; i < 256; ++i)
+                scEnv.close();
+
+            // try to delete scAlice, send funds to scBob
+            scEnv.tx(
+                acctdelete(scAlice, scBob),
+                fee(acctDelFee),
+                ter(tecHAS_OBLIGATIONS));
+        }
+    }
+
+    void
+    run() override
+    {
+        testXChainBridgeCreateConstraints();
+        return;
```

Comment:
I'm pretty confident that this `return` is left over from debugging and should be removed.  When you do that, you should probably also remove the `testXChainBridgeCreateConstraints()` in front of the `return`; it duplicates the call on line 4206.

Reply:
... And, once you remove the `return`, you'll need to patch up the `testXChainCreateBridge()` test a bit.